src/cli/common.rs

```diff
@@ -633,6 +634,22 @@ pub(crate) fn ignorable_error(
     }
 }
 
+/// Warns if rustup is trying to install a toolchain that might not be
+/// able to run on the host system.
+pub(crate) fn warn_if_host_is_incompatible(
+    toolchain: impl Display,
+    host_arch: &TargetTriple,
+    target_triple: &TargetTriple,
+    force_non_host: bool,
+) -> Result<()> {
+    if !force_non_host && !host_arch.can_run(target_triple)? {
+        error!("DEPRECATED: future versions of rustup will require --force-non-host to install a non-host toolchain.");
```

Comment:
I actually don't think this is a good way forward, since `host_arch.can_run()` relies on hardcoding and is fragile. The warnings below seem sufficient.

Reply:
Sounds good to me. Separate commit?

Reply:
@djc Let's get this change merged first and then discuss how to handle this in another issue/PR, since it looks like we already have some users who rely on this flag: https://github.com/search?q=%22--force-non-host%22&type=code